### PR TITLE
get request

### PR DESCRIPTION
**Pull request by timoshevichaleks**
Http get request

### CODE REVIEW

**Comment by timoshevichaleks** [comment]
first pull request

**Comment by dhutaryan** [comment]
General remarks:
- you don't have to call the branch with `branch` work. You can call just `posts`
- commit name `get request` doesn't anything. Commit message should say what you did. Like `add posts list` or `added posts list`
- The same about comment in description of pull request

File: src/app/app.module.ts [hunk 1]
```diff
@@ -3,14 +3,18 @@ import { BrowserModule } from '@angular/platform-browser';
 
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { JsonplaceholderComponent } from './jsonplaceholder/jsonplaceholder.component';
+import {HttpClient, HttpClientModule} from "@angular/common/http";
```
Comment (dhutaryan) [inline src/app/app.module.ts]: No spaces inside the brackets on the edges

File: src/app/data.service.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import {Post} from "./post";
+import {HttpClient, HttpHeaders, HttpParams} from "@angular/common/http";
+import {createI18nOptions} from "@angular-devkit/build-angular/src/utils/i18n-options";
```
Comment (dhutaryan) [inline src/app/data.service.ts]: Extra import

File: src/app/data.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DataService {
+
+  posts!: Post[];
+  siteUrl = `https://jsonplaceholder.typicode.com/posts`;
```
Comment (dhutaryan) [inline src/app/data.service.ts]: Move root url to `env` file

File: src/app/data.service.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import {Post} from "./post";
+import {HttpClient, HttpHeaders, HttpParams} from "@angular/common/http";
+import {createI18nOptions} from "@angular-devkit/build-angular/src/utils/i18n-options";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DataService {
+
+  posts!: Post[];
```
Comment (dhutaryan) [inline src/app/data.service.ts]: Why do you use `!` operator here?

File: src/app/data.service.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  siteUrl = `https://jsonplaceholder.typicode.com/posts`;
+
+  constructor(private http: HttpClient) { }
+
+  getData() {
```
Comment (dhutaryan) [inline src/app/data.service.ts]: Method name doesn't say anything

File: src/app/data.service.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    const url = this.siteUrl;
```
Comment (dhutaryan) [inline src/app/data.service.ts]: Extra code

File: src/app/data.service.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=UTF-8');
+    let params = new HttpParams();
```
Comment (dhutaryan) [inline src/app/data.service.ts]: Extra code

File: src/app/data.service.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+export class DataService {
+
+  posts!: Post[];
+  siteUrl = `https://jsonplaceholder.typicode.com/posts`;
+
+  constructor(private http: HttpClient) { }
+
+  getData() {
+    const url = this.siteUrl;
+    const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=UTF-8');
```
Comment (dhutaryan) [inline src/app/data.service.ts]: I think you shouldn't need that

File: src/app/data.service.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    let params = new HttpParams();
+
+    const options = { headers, params };
+
+    this.http.get<Post[]>(url, options).subscribe(
```
Comment (dhutaryan) [inline src/app/data.service.ts]: Unsubscribe from this observable

File: src/app/data.service.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+      res => {
+        this.posts = res
+        console.log(res)
```
Comment (dhutaryan) [inline src/app/data.service.ts]: Remove `console.log`

File: src/app/jsonplaceholder/jsonplaceholder.component.html [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+  <div class="row">
+    <div class="col">
+      <h2 class="py-3 text-center">POSTS</h2>
+    </div>
+  </div>
+</div>
+
+<div class="container">
+  <div class="row">
+    <div *ngFor="let post of this.data.posts">
```
Comment (dhutaryan) [inline src/app/jsonplaceholder/jsonplaceholder.component.html]: Is it work? I mean with `this`

Comment (dhutaryan) [reply src/app/jsonplaceholder/jsonplaceholder.component.html]: Anyway, don't use injectable service in the template directly

File: src/app/jsonplaceholder/jsonplaceholder.component.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import { Component, OnInit } from '@angular/core';
+import {DataService} from "../data.service";
+import {Post} from "../post";
+
+@Component({
+  selector: 'app-jsonplaceholder',
```
Comment (dhutaryan) [inline src/app/jsonplaceholder/jsonplaceholder.component.ts]: Don't see this component in the template anywhere

**Review by dhutaryan — CHANGES_REQUESTED** [review]
And it would be better to create a separate module for posts

File: src/app/data.service.ts (hunk 4 above)
Comment (timoshevichaleks) [reply src/app/data.service.ts]: Because variable initially null or undefind

File: src/app/data.service.ts (hunk 3 above)
Comment (timoshevichaleks) [reply src/app/data.service.ts]: moved url to root (enviroments/enviroment.ts)

File: src/app/data.service.ts (hunk 9 above)
Comment (timoshevichaleks) [reply src/app/data.service.ts]: replaced with async pipe

File: src/app/posts-module/posts/posts.component.ts [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+@Component({
+  selector: 'app-posts',
+  templateUrl: './posts.component.html',
+  styleUrls: ['./posts.component.scss']
+})
+export class PostsComponent implements OnInit {
+
+  posts$: Observable<Post[]>;
+
+  constructor(public postsService: PostsService) {
```
Comment (dhutaryan) [inline src/app/posts-module/posts/posts.component.ts]: Make it `private`

File: src/app/posts-module/posts/posts.component.html [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+
+
+<div class="container">
+  <div class="row">
+    <div class="col-6 mb-3" *ngFor="let post of posts$ | async">
+      <div class="card">
+        <div class="card-body">
+          <h5 class="card-title text-bg-primary">{{post.id}}. {{post.title | uppercase}}</h5>
+          <p class="card-text">{{post.body}}</p>
+          <a href="#" class="btn btn-primary">Go somewhere</a>
```
Comment (dhutaryan) [inline src/app/posts-module/posts/posts.component.html]: What this button does?

**Review by dhutaryan — CHANGES_REQUESTED** [review]

**Review by dhutaryan — APPROVED** [review]

**Pull request merged**